nixos/modules/services/torrent/transmission.nix

```diff
@@ -100,51 +101,113 @@ in
         default = "transmission";
         description = "Group account under which Transmission runs.";
       };
+
+      credentialsFile = mkOption {
+        type = types.path;
+        description = ''
+          Path to a JSON file to be merged with the settings.
+          Useful to merge a file which is better kept out of the Nix store
+          because it contains sensible data like <literal>rpc-password</literal>.
+        '';
+        default = "/dev/null";
+        example = "/var/lib/secrets/transmission/settings.json";
+      };
+
+      enableSandbox = mkOption {
```

Comment:
Oh no... this again? I thought this was meant as a temporary measure for `nginx` and will be removed before next stable release. This option is a bad idea.

Reply:
Can you elaborate why? I probably missed the discussion.While it may not be a pretty solution, it doesn't seem to be wrong to "test" the sandbox config with a simple option to switch back (though I'd probably prefer to have `true` as default).

Reply:
Citation for removing `enableSandbox` before next stable release: https://github.com/NixOS/nixpkgs/pull/85567#discussion_r423768923

@Ma27 if we add `enableSandbox` for this module, then why not for every other module? Who decides what `enableSandbox` means for a given module? How opinionated are we? Maybe we then start to need different ["levels" of sandboxing](https://github.com/NixOS/nixpkgs/pull/87661) adding another abstraction on NixOS, which in my opinion doesn't provide value. To clarify the "doesn't provide value" part of my last statement - utilizing `systemd` sandboxing features provides plenty of value, but arbitrarily choosing which sandboxing features and hiding it behind an `enableSandboxing` option makes this incredibly inconsistent and confusing for the user.

What should we do, at least in my opinion? Definitely apply every and all sandboxing option which has no negative impact to any conceivable workflow of a module. I see we have `NoNewPrivileges = true;` unconditionally set, which is great because that is an obvious benefit which won't conceivably impact any users, just protect their system. `PrivateDevices` and `PrivateTmp` are other great examples of sandboxing options which shouldn't impact users of `transmission` negatively. In fact most the options applied here, except `ProtectHome` seem entirely beneficial with no drawback. So what do we do for `ProtectHome`? I would say it is largely beneficial to apply that, so we should apply it and document that so the user knows.

Anyways... this is all just **my** opinion. As always, feel free to ignore :man_shrugging:

Reply:
I agree with @aanderse here. Sandboxing is not a boolean, but something very fuzzy.

Most of the options make sense in general. Those where an opt-out make sense with `mkDefault`, so opting-out becomes easier.

Reply:
I've removed this `enableSandbox` using `systemd-analyze security transmission` and set all the options accordingly to get the best score while maintaining transmission working.
I've also updated the apparmor profile which was missing paths specific to transmission, but also paths normally set in <abstractions/base> but which are not yet patched to fit NixOS. Leaving these paths until they become no longer needed if another PR makes AppArmor better integrated into NixOS.

Reply:
> if another PR makes AppArmor better integrated into NixOS.

I'm proposing https://github.com/NixOS/nixpkgs/pull/93457 for that.